### PR TITLE
Add vertx recipe based on CentOS

### PR DESCRIPTION
**Pull request by dharmit**
Helps take [eclipse/che PR #3843](https://github.com/eclipse/che/pull/3843) further.

cc: @l0rd

### CODE REVIEW

File: recipes/centos_vertx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+MAINTAINER Clement Escoffier
+
+ARG JAVA_VERSION=1.8.0
+ARG VERTX_VERSION=3.3.3
+
+RUN yum -y update && \
+    yum -y install sudo openssh-server centos-release-scl && \
+    yum clean all
+
+RUN yum -y install rh-maven33 java-${JAVA_VERSION}-openjdk-devel && \
```
Comment (benoitf) [inline recipes/centos_vertx/Dockerfile]: can we group all the RUN instructions ? (squashing layers during build is still experimental)

File: recipes/centos_vertx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  yum clean all
+
+ENV VERTX_GROUPID io.vertx
+ENV JAVA_HOME /usr/lib/jvm/java-${JAVA_VERSION}
```
Comment (benoitf) [inline recipes/centos_vertx/Dockerfile]: can we have a single ENV instruction ?

File: recipes/centos_vertx/cccp.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+job-id: vertx
```
Comment (benoitf) [inline recipes/centos_vertx/cccp.yml]: do we need that file in that folder ?

File: recipes/centos_vertx/install-vertx-dependencies.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
```
Comment (benoitf) [inline recipes/centos_vertx/install-vertx-dependencies.sh]: There is a missing copyright

File: recipes/centos_vertx/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd && \
+    echo "user ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
+    sed -i 's/Defaults    requiretty/#Defaults    requiretty/g' /etc/sudoers && \
+    useradd -u 1000 -G users,wheel -d /home/user --shell /bin/bash -m user && \
+    echo -e "secret\nsecret" | passwd user
+USER user
+
+COPY install-vertx-dependencies.sh /tmp/
+RUN scl enable rh-maven33 /tmp/install-vertx-dependencies.sh
```
Comment (benoitf) [inline recipes/centos_vertx/Dockerfile]: the install script could be removed at the end

File: recipes/centos_vertx/cccp.yml (hunk 3 above)
Comment (l0rd) [inline recipes/centos_vertx/cccp.yml]: @benoitf that file is needed for the centOS container pipeline to build it. @dharmit As for PR https://github.com/eclipse/che/pull/2643#issuecomment-253166429 can you add a comment that explains what is this file for?

**Comment by l0rd** [comment]
@dharmit please address @benoitf comments and it will be all set to merge.

File: recipes/centos_vertx/cccp.yml (hunk 3 above)
Comment (benoitf) [inline recipes/centos_vertx/cccp.yml]: yes but is it required in each folder or for example at the root of the repository ?

Comment (l0rd) [inline recipes/centos_vertx/cccp.yml]: Mmmh I'm not sure. I think we need one for every build job (i.e. one for every Dockerfile). @dharmit can you please confirm that?

Comment (dharmit) [inline recipes/centos_vertx/cccp.yml]: @l0rd @benoitf we need one for each recipe. Say you want to build a `registry.centos.org/che-stacks/vertx` and `registry.centos.org/che-stacks/springboot`, you'll have separate dirs for their Dockerfiles and each dir needs to have a relevant `cccp.yml`. 

Something like:

```
├── springboot
│   ├── cccp.yml
│   └── Dockerfile
└── vertx
    ├── cccp.yml
    └── Dockerfile
```

**Comment by dharmit** [comment]
@benoitf I'm not sure why it's taking `dshah@fedora.redhat.com`. In this case `fedora` is the hostname of the system I pushed from. However, when I did `git commit -s --amend --author "Dharmit Shah <dshah@redhat.com>"` I removed any references of `fedora` from it.

**Comment by dharmit** [comment]
Hi @benoitf, I think your requested changes have been taken care of. Besides that, I've moved `cccp.yml` to `.cccp.yml` as we did earlier [here](https://github.com/eclipse/che/pull/2643). Also, I overcame the sign-off issue I was facing earlier. Can you pleast take a look again?

cc: @l0rd

**Pull request merged**